src/server/routes/api/document/indra.js

```diff
@@ -133,7 +144,10 @@ const getDocuments = ( templates, queryDoc ) => {
         queryText = handleNoQueryAbastract( queryDoc );
       }
 
-      let queryUid = getPmid( queryArticle );
+      // semantic search api requires uid for query but there is no
+      // use case of it at least for us. Therefore, it must be
+      // fine to generate a random uuid when the pmid is not available.
```

Comment:
You're right. At the time, I thought semantic-search service could be responsible for fetching article Abstract/text and the API would be a list of query and document `uids` only which would be more lightweight.

Reply:
That would be ideal.  That would be good to have after improving the results of the related papers system -- so that it doesn't return zero results or one result much of the time.